app/models/technology.rb

```diff
@@ -70,7 +70,11 @@ def self.generic
   # Public: Retrieves the record with the matching +key+ or raises
   # ActiveRecord::RecordNotFound if no such record exists.
   def self.by_key(key)
-    key == 'generic' ? generic : where(key: key).first
+    if technology = where(key: key).first
+      technology
+    else
+      raise ActiveRecord::RecordNotFound, "can't find Technology with type #{ key }"
+    end
```

Comment:
This could be simplified as...

```ruby
Technology.find_by_key!(key)
```

In fact, that's *so* simple we could replace all calls to `Technology.by_key` with `find_by_key!`?